go/worker/common/committee/node.go

```diff
@@ -432,10 +433,21 @@ func (n *Node) updateHostedRuntimeVersionLocked() {
 		activeVersion = activeDeploy.Version
 	}
 
-	if err := n.SetHostedRuntimeVersion(n.ctx, activeVersion); err != nil {
-		n.logger.Error("failed to activate runtime version",
+	// For compute nodes, determine if there is a next version and activate it early.
+	var nextVersion *version.Version
+	if config.GlobalConfig.Mode == config.ModeCompute {
+		nextDeploy := n.CurrentDescriptor.NextDeployment(n.CurrentEpoch)
+		if nextDeploy != nil {
```

Comment:
```suggestion
		if nextDeploy != nil && nextDeploy.ValidFrom - n.CurrentEpoch < 12345 {
```
We probably don't want to start doing attestations to early.

Reply:
Agreed, made this configurable, please take a look.